sys/drv/bcm2835_pic.c

```diff
@@ -0,0 +1,204 @@
+#define KL_LOG KL_DEV
+#include <sys/bus.h>
+#include <sys/devclass.h>
+#include <sys/errno.h>
+#include <sys/fdt.h>
+#include <sys/interrupt.h>
+#include <sys/klog.h>
+#include <sys/libkern.h>
+#include <dev/bcm2835reg.h>
+
+/*
+ * located at BCM2835_ARMICU_BASE
+ * accessed by BCM2835_PERIPHERALS_BASE NOT by BCM2835_PERIPHERALS_BASE_BUS
+ * 32 GPU0 interrupts -- 0x204 offset from address; 0 in table
+ * 32 GPU1 interrupts -- 0x208 offset from address; 32 in table
+ * 32 basic interrupts -- 0x200 offset from address; 64 in table
+ */
+
+typedef enum bcm2835_intrtype {
+  BCM2835_INTRTYPE_BASIC,
+  BCM2835_INTRTYPE_GPU0,
+  BCM2835_INTRTYPE_GPU1,
+  BCM2835_INTRTYPE_CNT
+} bcm2835_intrtype_t;
+
+#define BCM2835_NIRQPERTYPE (BCM2835_NIRQ / BCM2835_INTRTYPE_CNT)
+
+typedef struct bcm2835_pic_state {
+  rman_t rm;
+  resource_t *mem;
+  resource_t *irq;
+  intr_event_t *intr_event[BCM2835_NIRQ];
+} bcm2835_pic_state_t;
+
+#define in4(addr) bus_read_4(bcm2835_pic->mem, (addr))
+#define out4(addr, val) bus_write_4(bcm2835_pic->mem, (addr), (val))
+
+static void bcm2835_pic_disable_irq(intr_event_t *ie) {
+  bcm2835_pic_state_t *bcm2835_pic = ie->ie_source;
+  unsigned irq = ie->ie_irq;
+
+  bus_size_t reg = BCM2835_INTC_IRQBDISABLE;
+  if (irq < BCM2835_INT_GPU1BASE)
+    reg = BCM2835_INTC_IRQ1DISABLE;
+  else if (irq < BCM2835_INT_BASICBASE)
+    reg = BCM2835_INTC_IRQ2DISABLE;
```

Comment:
Minor detail, I just found the ordering of these ifs to be confusing.
```suggestion
  bus_size_t reg = 0;
  if (irq < BCM2835_INT_GPU1BASE)
    reg = BCM2835_INTC_IRQ1DISABLE;
  else if (irq < BCM2835_INT_BASICBASE)
    reg = BCM2835_INTC_IRQ2DISABLE;
  else
    reg = BCM2835_INTC_IRQBDISABLE;
```

Reply:
I considered multiple solutions but all of them are equal to current code. It's short, clean, easy to understand. You can add brackets for better readability.

I don't like @mohrcore solution because it's more confusing to reader. You add another branch. `reg` is explicitly initialized for 0 without any reason - it's another level of confusion. As reviewer I can think that you forgot about something and your code is invalid because of that additional initialization.